clients/venice-admin-tool/src/test/java/com/linkedin/venice/TestAdminTool.java

```diff
@@ -164,4 +165,33 @@ public void testAdminTopicIsAllowedByTopicConfigsRelatedApi() {
       Assert.fail("AdminTool should allow admin topic to be queried by config query API", e);
     }
   }
+
+  @Test
+  public void testAdminToolDataRecoveryApi() {
+    String storeNames = "test1,test2,test3";
+
+    String[] estimateArgs = { "--estimate-data-recovery-time", "--url", "http://localhost:7036", "--stores", storeNames,
+        "--dest-fabric", "ei-ltx1" };
+
+    String[] executeArgs = { "--execute-data-recovery", "--recovery-command", "venice-tools", "--extra-command-args",
+        "repush kafka --force --format", "--url", "http://localhost:7036", "--stores", storeNames, "--source-fabric",
+        "ei4", "--dest-fabric", "ei-ltx1", "--non-interactive", "--datetime", "2023-06-27T14:19:25" };
+
+    String[] monitorArgs = { "--monitor-data-recovery", "--url", "http://localhost:7036", "--stores", storeNames,
+        "--dest-fabric", "ei-ltx1", "--datetime", "2023-06-27T14:19:25", "--interval", "300" };
+
+    String[] estimateArgs2 = { "--estimate-data-recovery-time", "--url", "http://localhost:7036", "--cluster",
+        "venice-1", "--dest-fabric", "ei-ltx1" };
+
+    try {
+      AdminTool.main(estimateArgs);
+      AdminTool.main(estimateArgs2);
+      AdminTool.main(executeArgs);
+      AdminTool.main(monitorArgs);
+    } catch (VeniceClientException e) {
+      // Expected exception.
+    } catch (Exception err) {
+      Assert.fail("AdminTool should allow admin topic to be updated by config update API", err);
+    }
```

Comment:
1. When will VeniceClientException throw?
2. The assertion fail message does not sound right.

Reply:
1. This is from the new code that I added in this change.
```
    if (!StringUtils.isEmpty(cluster)) {
      try (ControllerClient parentCtrlCli = new ControllerClient(cluster, parentUrl, sslFactory)) {
        MultiStoreResponse response = parentCtrlCli.queryStoreList(false);
        if (response.isError()) {
          throw new VeniceClientException("Error in getting store list for " + cluster + ": " + response.getError());
        }
        if (response.getStores() == null) {
          return null;
        }
        return new HashSet<>(Arrays.asList(response.getStores()));
      }
    }
```
It would throw the following exception if I don't catch it, when running locally.
```
Caused by: com.linkedin.venice.client.exceptions.VeniceClientException: Error in getting store list for venice-1: An error occurred during controller request. controller = null, route = /list_stores, params = [cluster_name=venice-1, hostname=localhost, include_system_stores=false], timeout = 600000. Exception type: class com.linkedin.venice.exceptions.VeniceException. Detailed message: Unable to discover leader controller from [http://localhost:7036]
	at com.linkedin.venice.AdminTool.calculateRecoveryStoreNames(AdminTool.java:744)
	at com.linkedin.venice.AdminTool.estimateDataRecoveryTime(AdminTool.java:674)
	at com.linkedin.venice.AdminTool.main(AdminTool.java:518)
	at com.linkedin.venice.TestAdminTool.testAdminToolDataRecoveryApi(TestAdminTool.java:188)
```

2. Oops, yeah, it's a copy-and-paste error. Will fix it.